### ISSUE TITLE
get_feature does not filter feature by parcellation

### ISSUE THREAD

**Issue by xgui3783** [FZJ-INM1-BDA/siibra-python#24]
after `select_parcellation`, `get_feature` is expected to get features from the selected parcellation. 0.0.9.dev3 currently does not.

```
>>> import siibra as sb
[siibra:INFO]  Selected parcellation "Julich-Brain Cytoarchitectonic Maps 2.5"
[siibra:INFO]  Version: 0.0.9.dev3
>>> atlas = sb.atlases.MULTILEVEL_HUMAN_ATLAS
>>> atlas.select_parcellation(sb.parcellations['minds/core/parcellationatlas/v1.0.0/94c1125b-b87e-45e4-901c-00daee7f2579-25'])
[siibra:INFO]  Selected parcellation "Julich-Brain Cytoarchitectonic Maps 2.5"
>>> atlas.selected_parcellation.id
'minds/core/parcellationatlas/v1.0.0/94c1125b-b87e-45e4-901c-00daee7f2579-25'
>>> atlas.select_region('hoc1 left')
[siibra:INFO]  Selected region Area hOc1 (V1, 17, CalcS) - left hemisphere
Area hOc1 (V1, 17, CalcS) - left hemisphere
>>> conn=atlas.get_features(sb.modalities.ConnectivityProfile)
>>> conn[1].parcellation.id
'minds/core/parcellationatlas/v1.0.0/94c1125b-b87e-45e4-901c-00daee7f2579'
>>> conn[1].parcellation.name
'Julich-Brain Cytoarchitectonic Maps 1.18'
>>> 
```

**Comment by xgui3783** [comment]
@dickscheid do you have time to take a look at this? I can provide a PR if you would like

**Comment by dickscheid** [comment]
I will have a look.

**Comment by dickscheid** [comment]
I can confirm the problem. I will provide a fix. Unfortunately, implementing the fix reveals a minor inconsistency in the connectivity dataset that I need to investigate.

**Comment by dickscheid** [comment]
I push a fix to develop now., which fixes the problem. Unfortunately the ConnectivityProfileExtractor is quite slow now, because every column name in the matrix is tested to be a valid region for each 'get_features' call of the atlas object. NEed to look into optimizing this.

**Comment by xgui3783** [comment]
@dickscheid can you check https://github.com/FZJ-INM1-BDA/siibra-python/commit/8fe7c9ba80c1af8cd02901e2ba565b7bbbe81a8c , on first glance on my part, it seems to filter the conn profile correctly

and it does not seem to have much impact on perf

**Comment by xgui3783** [comment]
fixed by #26
